content/src/main/content/jcr_root/apps/acs-commons/touchui-widgets/showhidedialogfields/source/requiredvalidation.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * Register required field validation.
+ * 
+ * The validator keys off of the ".field-required" selector. To use in Touch UI
+ * dialogs, add the class "field-required" to a textfield. It is implemented to
+ * get around the problem that a hidden required field cannot be submitted. The
+ * default AEM validator validates the field even if it is hidden.
+ * 
+ */
+$.validator
```

Comment:
`$.validator` is deprecated. This should use `foundation.validation.validator` instead (and IIUC the code will be a lot shorter as a result).

Reply:
change it to use _acs-commons-field-required-allow-hidden_